ctrd/container.go

```diff
@@ -122,14 +122,13 @@ func (c *Client) execContainer(ctx context.Context, process *Process) error {
 			exitTime: status.ExitTime(),
 		}
 
-		if err := <-fail; err != nil {
-			msg.err = err
-		}
-
-		for _, hook := range c.hooks {
-			if err := hook(process.ExecID, msg); err != nil {
-				logrus.Errorf("failed to execute the exec exit hooks: %v", err)
-				break
+		// run hook if not got fail here
+		if err := <-fail; err == nil {
```

Comment:
can we get the real exit code?

Reply:
When fails here, we got a empty(meaningless) exit status, that means exit code is zero.